universal-application-tool-0.0.1/app/auth/IdcsProfileAdapter.java

```diff
@@ -28,6 +28,40 @@ protected String emailAttributeName() {
     return "user_emailid";
   }
 
+  @Override
+  public UatProfileData mergeUatProfile(UatProfile uatProfile, OidcProfile oidcProfile) {
+    String locale = oidcProfile.getAttribute("user_locale", String.class);
+    List<CompletionStage<Void>> dbOperations = List.of();
+    if (locale != null && !locale.isEmpty()) {
+      dbOperations.add(
+          uatProfile
+              .getApplicant()
+              .thenApplyAsync(
+                  applicant -> {
+                    applicant.getApplicantData().setPreferredLocale(Locale.forLanguageTag(locale));
+                    applicant.save();
+                    return null;
+                  }));
+    }
+    String displayName = oidcProfile.getAttribute("user_displayname", String.class);
+    if (displayName != null && !displayName.isEmpty()) {
+      dbOperations.add(
+          uatProfile
+              .getApplicant()
+              .thenApplyAsync(
+                  applicant -> {
+                    applicant.getApplicantData().setUserName(displayName);
+                    applicant.save();
+                    return null;
+                  }));
+    }
+    for (CompletionStage<Void> dbOp : dbOperations) {
+      dbOp.toCompletableFuture().join();
+    }
```

Comment:
suggestion (non-blocking): If we don't care about being async here, we could complete the futures in the if statements instead

Reply:
This is *marginally* more parallel in that the first request can be completed while we're sending the second.  Barely worth it, but if we wind up with a ton of these for one of the identity providers, it might be a good pattern.

Reply:
sounds good to me!